crates/nu-command/src/conversions/into/filesize.rs

```diff
@@ -148,29 +149,35 @@ fn int_from_string(a_string: &str, span: Span) -> Result<i64, ShellError> {
     let clean_string = no_comma_string.trim();
 
     // Hadle negative file size
-    if let Some(stripped_string) = clean_string.strip_prefix('-') {
-        match stripped_string.parse::<bytesize::ByteSize>() {
+    if let Some(stripped_negative_string) = clean_string.strip_prefix('-') {
+        match stripped_negative_string.parse::<bytesize::ByteSize>() {
             Ok(n) => Ok(-(n.as_u64() as i64)),
```

Comment:
Are there any situations where casting may fail?

Reply:
Yes, there are. If you want, feel free to fix this as well, but it's not required.

If you do want to fix it on both sides, I would suggest using `try_into` or `i64::try_from`, followed by `map_err()` and reusing your `string_to_into_convert_error` function (the same error message is fine).

Reply:
I see. Can i ask why you think it is not required? Because it's not related to this issue directly. Or because it is difficult to actually happen?

Reply:
I got it! your suggestion need implementation `TryInto` traits to ByteSize type. As you said, I agree the types of work are little different. I will make new PR to apply your upper suggestion.

Reply:
Thanks! Also, I should have clarified. I think we have to use `i64::try_from(0_u64)` (which is the same as `0_u64.try_into()`). The `ByteSize` type is from an external crate, and I don't think we can add trait implementations to it unfortunately.